public/friendlycode/js/fc/filesystem-sync.js

```diff
@@ -0,0 +1,104 @@
+define(["jquery"], function($) {
+  "use strict";
+
+  var FileSystemSync = {};
+
+  function pushFileChange(url, csrfToken, fs, path) {
+    var options = {
+      contentType: "application/json",
+      headers: {
+        "X-Csrf-Token": csrfToken
+      },
+      type: "PUT",
+      url: url
+    };
+
+    function send() {
+      var request = $.ajax(options);
+      request.done(function() {
+        if(request.readyState !== 4) {
+          return;
+        }
+
+        if(request.status !== 201 && request.status !== 200) {
+          // TODO: handle error case here
+          console.error("Server did not persist file");
+          return;
+        }
+
+        console.log("Successfully persisted ", path);
```

Comment:
I have an object called `fsync` that I think should be emitting these events. But I think we should wait until @sedge's refactor so that we can use node's `EventEmitter` to make our lives easier.